File: 01_introduccion/ejemplos/suma_lista.c
// Implementacion en C, desde un enfoque imperativo, del programa que calcula la suma de una lista de numeros

#include <stdio.h>

int sum(int array[], int size) {
    int total = 0; 
    for (int i = 0; i < size; i++) {
        total += array[i];
    }
    return total;
}

int main() {
    int numbers[] = {1, 2, 3, 4, 5};
    int size = sizeof(numbers) / sizeof(numbers[0]);
    
    printf("Sum: %d\n", sum(numbers, size));  // Salida: 15
    return 0;
}

File: 01_introduccion/ejemplos/suma_lista.erl
%% Archivo: sum_list.erl
-module(sum_list).
-export([sum/1, main/1]).

% Suma los elementos de una lista.
sum([]) -> 0;  % Caso base: si la lista está vacía, la suma es 0.
sum([Head | Tail]) -> Head + sum(Tail).  % Caso recursivo: sumar el primer elemento al resultado de la suma del resto de la lista.

% Función principal, que se encarga de invocar la lógica y mostrar el resultado.
main(_Args) ->
    List = [1, 2, 3, 4, 5],  % La lista de números.
    Result = sum(List),  % Llamamos a la función sum para obtener el resultado.
    io:format("Suma: ~p~n", [Result]).  % Imprimir el resultado.
